tests/training/features/factors/setup.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash -x 
+
+##################################################################### 
+# AUTHOR: pedrodiascoelho
+#####################################################################
+
+# Exit on error
+set -e
+
+# Test code goes here
+test -f $MRT_DATA/europarl.de-en/toy.bpe.en || exit 1
+test -f $MRT_DATA/europarl.de-en/toy.bpe.de || exit 1
+
+#escape carachters #:_\| 
+test -s toy.bpe.esc.en || cat $MRT_DATA/europarl.de-en/toy.bpe.en | sed 's/#/\&htg;/g;s/:/\&cln;/g;s/_/\&usc;/g;s/\\/\&esc;/g;s/|/\&ppe;/g' \
+                                        > toy.bpe.esc.en
+test -s toy.bpe.esc.de ||  cat $MRT_DATA/europarl.de-en/toy.bpe.de | sed 's/#/\&htg;/g;s/:/\&cln;/g;s/_/\&usc;/g;s/\\/\&esc;/g;s/|/\&ppe;/g' \
+                                        > toy.bpe.esc.de
+
+#add factors to replace @@ markers. s1 is used if a word is a subword (if it has the suffix @@), s0 is used otherwise
+test -s toy.bpe.fact.en || cat toy.bpe.esc.en | sed 's/\(\s\|$\)/|s0 /g;s/@@|s0/|s1/g;s/\s*$//' > toy.bpe.fact.en
+test -s toy.bpe.fact.de || cat toy.bpe.esc.de | sed 's/\(\s\|$\)/|s0 /g;s/@@|s0/|s1/g;s/\s*$//' > toy.bpe.fact.de
+
+#creates factored vocabulary
+if [[ ! -s vocab.en.fsv ]]; then
```

Comment:
For simplicity and to improve the readability, can we add `vocab.{en,de}.fsv` files to the repo instead of generating them on the fly?

Reply:
Yes of course, that simplifies a lot the setup.sh file. I'll just use the ones that are currently in the repo in the models folder, for the decoding test. This way I don't even need to escape the characters and I can remove that for simplicity. They would only give problems when constructing the vocabulary.